### ISSUE TITLE
Impossible to create neuron with both excitatory and inhibitory connections

### ISSUE THREAD

**Issue by Kobrs** [UCI-CARL/CARLsim3#38]
It looks like there is no way of creating setup where single neuron connects to two others, where one synapse is excitatory, while the second one inhibitory. This is due to the fact that connection type for group is always established based on presynaptic neuron.

**Comment by mbeyeler** [comment]
You're right about that. However, this is not a mere limitation of CARLsim, but is generally true in the brain. It's called [Dale's principle](https://en.wikipedia.org/wiki/Dale%27s_principle), and states that a neuron "performs the same chemical action at all of its synaptic connections to other cells, regardless of the identity of the target cell".

**Comment by Kobrs** [comment]
Yes, but this means that neuron releases single type of neurotransmitter in all its synapses (actually this is often not true, because neuron can release more than one neurotransmitter), but often single neurotransmitter can have opposing effect on two postsynaptic cells. For example presynaptic neuron could release GABA but one of postsynaptic neurons have inhibitory receptor while the second excitatory.
